lib/std/os.zig

```diff
@@ -2611,7 +2611,7 @@ pub fn realpathC(pathname: [*:0]const u8, out_buffer: *[MAX_PATH_BYTES]u8) RealP
 
         return readlinkC(@ptrCast([*:0]const u8, proc_path.ptr), out_buffer);
     }
-    const result_path = std.c.realpath(pathname, out_buffer) orelse switch (std.c._errno().*) {
+    const result_path = std.c.realpath(pathname, @ptrCast([*:0]u8, out_buffer)) orelse switch (std.c._errno().*) {
```

Comment:
I think the type of `out_buffer` should change rather than having a `@ptrCast`?

Reply:
I actually thought it made more sense to change the signature of realpath, since being null terminated is only a post-condition, does that sound ok?

Reply:
Also, you then get the null terminated slice as the return value